### PR TITLE
MWPW-121471 Layout in section metadata

### PR DESCRIPTION
**Pull request by JasonHowellSlavin**
* Adding layout to section metadata, and css for new class additions
* Adding test to account for new function

Resolves: [MWPW-121471](https://jira.corp.adobe.com/browse/MWPW-121471)

**Test URLs:**
- Before: https://main--bacom--adobecom.hlx.page/drafts/slavin/layout-sample?martech=off
- After: https://main--bacom--adobecom.hlx.page/drafts/slavin/layout-sample?martech=off&milolibs=section-metadata-layout

- After:On Milo, with content originally set up by @ryanmparrish: https://section-metadata-layout--milo--adobecom.hlx.page/drafts/slavin/document

### CODE REVIEW

File: libs/blocks/section-metadata/section-metadata.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ function handleStyle(div, section) {
   }
 }
 
+function handleLayout(div, section) {
+  const layoutString = div.textContent.trim();
+  const layoutClass = `grid-template-columns-${layoutString.replaceAll(' | ', '-')}`;
+
+  if (section) {
+    section.classList.add(layoutClass);
+  }
```
Comment (chrischrischris) [inline libs/blocks/section-metadata/section-metadata.js:29]: nit
```suggestion
  if (!(div || section)) return;
  const layoutString = div.textContent.trim();
  const layoutClass = `grid-template-columns-${layoutString.replaceAll(' | ', '-')}`;
  section.classList.add(layoutClass);
```

File: libs/blocks/section-metadata/section-metadata.css [hunk 2]
```diff
@@ -183,4 +183,19 @@ main>.section[class*='-up']>.content {
...
+    grid-template-columns: 2fr 1fr;
+  }
+
+  .section.grid-template-columns-1-3 {
+    grid-template-columns: 1fr 3fr;
+  }
+  
+  .section.grid-template-columns-3-1 {
+    grid-template-columns: 3fr 1fr;
+  }
```
Comment (rgclayton) [inline libs/blocks/section-metadata/section-metadata.css:200]: I love the use of `fr`unit. My only concern here is related to the `grid-template-columns` and all of the possible variations needed to support a full 12 col grid system in the future. Would Flexbox be more straightforward? Thoughts?

**Review by chrischrischris — APPROVED** [review]

File: libs/blocks/section-metadata/section-metadata.css (hunk 2 above)
Comment (JasonHowellSlavin) [reply libs/blocks/section-metadata/section-metadata.css:200]: I believe with Flexbox we would have to have child items of the parent declare their widths. So we'd have to add classes to children in the code and additional classes for widths like `span-4-col` right?.

With the `grid-template-columns` the parent container controls the children, and we already have the makings of a system with classes like `two-up` and `three-up`. But to your point, we may haver to add quite a few classes to account for variants. Perhaps if we get to that point we could actually construct the value for `grid-template-columns` with JS?

Comment (rgclayton) [reply libs/blocks/section-metadata/section-metadata.css:200]: >I believe with Flexbox we would have to have child items of the parent declare their widths. So we'd have to add classes to children in the code and additional classes for widths like span-4-col right?.

Yeah, i think there is some give and take here. I just wanted to throw the thought out there for people to think about.

**Review by rgclayton — APPROVED** [review]

**Pull request merged**